kong/cmd/utils/migrations.lua

```diff
@@ -60,6 +60,10 @@ local function up(schema_state, db, opts)
     error("Cannot run migrations: " .. NEEDS_BOOTSTRAP_MSG)
   end
 
+  if not opts.force and schema_state.pending_migrations then
+    error("Database has pending migrations; run 'kong migrations finish'")
+  end
```

Comment:
Why changing the below `error` call to `log`, like all other logs written from the cluster mutex?

Reply:
@thibaultcha, it will change the status code then too that cli returns. I think that would be breaking.

Reply:
Ok - should it be removed then? They just seem redundant so the diff looked incomplete to me, that's all!

Reply:
The repetitive test is there because the test I added here is ran without a lock. So there is tiny change that the first check passes while the second will fail after waiting for the mutex. The one inside the mutex will still result stack trace. We can also gather that to `err_without_stack` and then throw the `err_without_stack` after returning from the code running inside mutex. But that looked so strange, so I just copied the check. This will be ok for 99,999% of the time (people should never hit the one inside a mutex, but there is a small change). The errors raised inside the mutex contain stack trace. That is deeper and I didn't want to make change there (e.g. a new option to not add stacktrace).

Reply:
What about:

`lapp.quit("Database has pending migrations; run 'kong migrations finish", true)`? That way it will not be breaking and we avoid duplicating the checks just for cosmetic reasons (also it is slightly more robust and easier to understand).

Reply:
@dndx, whenever we do `quit` we do not release mutex then.

Reply:
Okay, if that is the case I guess this is the most reasonable fix we can do given the current implementation, I will be okay to merge this for now, and if we have a better way of propagating the error we can always change it in the future.

Can we add some comment before this change just so that future maintainers won't scratch their head figuring out why a double check was needed?